mbuild/lattice.py

```diff
@@ -0,0 +1,260 @@
+                                  'or <tuple>. {} was passed in as type: {} '
+                                  'from basis_vectors.'
+                                  .format(lst[1], type(lst[1])))
+                else:
+                    ValueError('Basis Vector does not match dimension. '
+                               '{} does not have vector components equal to '
+                               'the dimension specified: {}'
+                               .format(lst[1], dimension))
+        else:
+            TypeError('Incorrect Type: "basis_vectors" is of type {}, '
+                      'not type <list> or <tuple>. Please review '
+                      'the documentation for proper format.'
+                      .format(type(basis_vectors)))
+
+        # basis overlap check
+        """
+        check for two same basis values in dict already
+        then check if 2 vectors then overlap
+        error if that occurs
+        Achieved by expanding unit cell out 2 iterations
+        """
+        if isinstance(basis_vectors, defaultdict):
+            pass
+        else:
+            # test for overlap
+            overlap_dict = defaultdict(list)
+            for the_id, vector in basis_vectors:
+                if dimension == 3:
+                    for x in range(3):
+                        for y in range(3):
+                            for z in range(3):
+                                tmpx = vector[0] + x
+                                tmpy = vector[1] + y
+                                tmpz = vector[2] + z
+                                tmp_tuple = tuple((tmpx, tmpy, tmpz))
+                                value = tuple((the_id, vector))
+                                overlap_dict[tmp_tuple].append(value)
+
+                elif dimension == 2:
+                    for x in range(3):
+                        for y in range(3):
+                            tmpx = vector[0] + x
+                            tmpy = vector[1] + y
+                            tmp_tuple = tuple((tmpx, tmpy))
+                            value = tuple((the_id, vector))
+                            overlap_dict[tmp_tuple].append(value)
+
+                else:
+                    for x in range(3):
+                        tmpx = vector[0] + x
+                        tmp_tuple = tuple((tmpx))
+                        value = tuple((the_id, vector))
+                        overlap_dict[tmp_tuple].append(value)
+
+            # now review for overlap
+            for key, val in overlap_dict.items():
+                if len(val) > 1:
+                    ValueError('Overlapping Basis Vectors: Basis vectors '
+                               'overlap when the unit cell is '
+                               'expanded to {}. This is an incorrect perfect '
+                               'lattice. The vectors are: {}'
+                               .format(key, val))
+
+        return {'dim': dimension, 'lat_vec': lattice_vectors,
```

Comment:
You could set the attributes here instead of returning (as mentioned in comment above)